src/model/models/modelManager.ts

```diff
@@ -61,4 +65,44 @@ export class ModelManager {
       throw new AppError('', httpStatus.INTERNAL_SERVER_ERROR, 'store-trigger service is not available', true);
     }
   }
+
+  public async deleteModel(identifier: string): Promise<StoreTriggerResponse> {
```

Comment:
Refactor this function. 
You don't need to check twice if the record is undefined.
The getRecord  function should be inside the try block.